src/main/java/teammates/logic/backdoor/BackDoorLogic.java

```diff
@@ -439,8 +439,13 @@ public void removeDataBundle(DataBundle dataBundle) {
         }
         accountsDb.deleteAccounts(dataBundle.accounts.values());
 
-        // Delete all admin emails
-        adminEmailsDb.deleteEntities(adminEmailsDb.getAllAdminEmails());
+        for (AdminEmailAttributes email : dataBundle.adminEmails.values()) {
+            AdminEmailAttributes emailInDb = adminEmailsDb.getAdminEmailBySubject(email.subject);
+            if (emailInDb == null) {
```

Comment:
1. Is there a legit reason for this to be null? May be a comment to explain how this branch can be taken?
2. Is there a risk of deleting legit emails if they happen to have the same subject as an email used in testing?